monkey/monkey_island/cc/repository/file_agent_configuration_repository.py

```diff
@@ -12,10 +20,15 @@ def __init__(self, file_repository: IFileRepository):
         self._schema = AgentConfigurationSchema()
 
     def get_configuration(self) -> AgentConfiguration:
-        with self._file_repository.open_file(AGENT_CONFIGURATION_FILE_NAME) as f:
-            configuration_json = f.read().decode()
-
-        return self._schema.loads(configuration_json)
+        try:
+            with self._file_repository.open_file(AGENT_CONFIGURATION_FILE_NAME) as f:
+                configuration_json = f.read().decode()
+
+            return self._schema.loads(configuration_json)
+        # TODO: Handle FileRetrievalError vs FileNotFoundError
+        #       https://github.com/guardicore/monkey/blob/e8001d8cf76340e42bf17ff62523bd2d85fc4841/monkey/monkey_island/cc/repository/file_storage/local_storage_file_repository.py#L47-L50
+        except RetrievalError:
```

Comment:
Why not FileRetrievalError?

Reply:
As the TODO says, there's more work to be done here. The next PR will add a new error type and have it handle that instead.